fdbserver/BlobManager.actor.cpp

```diff
@@ -5660,14 +5653,33 @@ ACTOR Future<Void> truncateMutationLogs(Reference<BlobManagerData> bmData) {
 			state Version logEndVersion = desc.contiguousLogEnd.get();
 			TraceEvent("DescribedMutationLogs").detail("LogEndVersion", logEndVersion);
 
-			// Force flush in-memory data of all blob workers until end of log
-			FlushGranuleRequest req(bmData->epoch, normalKeys, logEndVersion, false);
-			wait(
-			    success(doBlobGranuleRequests(bmData->db, normalKeys, req, &BlobWorkerInterface::flushGranuleRequest)));
+			// Flush until end of the mutation log
+			state std::vector<KeyRange> flushRanges;
+			for (auto& it : bmData->knownBlobRanges.intersectingRanges(normalKeys)) {
+				if (it.cvalue()) {
+					flushRanges.push_back(it.range());
+				}
+			}
+			TraceEvent("FlushingBlobGranules").detail("Ranges", flushRanges.size());
+			for (auto& range : flushRanges) {
+				FlushGranuleRequest req(bmData->epoch, range, logEndVersion, false);
+				wait(success(doBlobGranuleRequests(bmData->db, range, req, &BlobWorkerInterface::flushGranuleRequest)));
```

Comment:
for performance it's going to be very desirable to do these in parallel when we have a large number of ranges

Reply:
fixed. thanks!